x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStartTrainedModelDeploymentAction.java

```diff
@@ -292,10 +302,16 @@ public boolean test(ClusterState clusterState) {
                 );
                 return true;
             }
+            Set<String> nodesShuttingDown = nodesShuttingDown(clusterState);
+            List<DiscoveryNode> nodes = clusterState.nodes()
+                .getAllNodes()
+                .stream()
+                .filter(d -> nodesShuttingDown.contains(d.getId()) == false)
+                .filter(TaskParams::mayAllocateToNode)
+                .collect(Collectors.toList());
 
             // No nodes allocated at all!
-            // TODO when we support autoscaling for this, check for `maxLazyNodes` setting
-            if (nodesAndState.isEmpty()) {
+            if (nodesAndState.isEmpty() && maxLazyMLNodes < nodes.size()) {
```

Comment:
What does it mean when `maxLazyMLNodes < nodes.size()`?
Shouldn't it rather be `maxLazyMLNodes <= nodes.size()`?

Sorry if it's obvious but I don't quite understand this condition.

Reply:
`<=` is indeed correct :)

Basically, `nodesAndState` indicates that no node has been added to the allocation when it was created. So, there were no valid nodes with room.

This is checking if there are "lazy" nodes, thus indicating autoscaling capabilities.